src/Workspaces/Core/Portable/Shared/Utilities/ProducerConsumer.cs

```diff
@@ -126,27 +145,26 @@ internal static class ProducerConsumer<TItem>
         TArgs args,
         CancellationToken cancellationToken)
     {
-        return RunAsync(
-            // We're running in parallel, so we def have multiple writers
-            ProducerConsumerOptions.SingleReaderOptions,
-            produceItems: static (callback, args, cancellationToken) =>
-                RoslynParallel.ForEachAsync(
-                    args.source,
-                    cancellationToken,
-                    async (source, cancellationToken) =>
-                        await args.produceItems(source, callback, args.args, cancellationToken).ConfigureAwait(false)),
-            consumeItems: static (enumerable, args, cancellationToken) => args.consumeItems(enumerable, args.args, cancellationToken),
-            args: (source, produceItems, consumeItems, args),
+        // Bridge to sibling helper that takes a consumeItems that returns a value.
+        return RunParallelAsync(
+            source,
+            produceItems: static (item, callback, args, cancellationToken) => args.produceItems(item, callback, args.args, cancellationToken),
+            consumeItems: static async (items, args, cancellationToken) =>
+            {
+                await args.consumeItems(items, args.args, cancellationToken).ConfigureAwait(false);
+                return default(VoidResult);
```

Comment:
similarly, the Task based mthods work by deferring to the Task<T> version, using VoidResult as the ignored type arg.